### PR TITLE
Create analysis from cart

### PR DESCRIPTION
**Pull request by squirrelo**
This adds the `from_default` flag to the Analysis object, allowing you to create a new analysis from previously selected samples. This also modifies the analysis creation to use queues. 

~~This pull request relies on #1038 so please merge that one first.~~ Merged. Many of the file changes are from that, only `Analysis.create()` and the `test_analysis_from_default` test function were modified in this pull request.

### CODE REVIEW

**Comment by squirrelo** [comment]
Ready for review

File: qiita_db/analysis.py [hunk 1]
```diff
@@ -112,23 +112,44 @@ def create(cls, owner, name, description, parent=None):
...
-               "analysis_status_id) VALUES (%s, %s, %s, 1) "
-               "RETURNING analysis_id".format(cls._table))
-        a_id = conn_handler.execute_fetchone(
-            sql, (owner.id, name, description))[0]
+        if from_default:
+            # insert analysis and move samples into that new analysis
+            dflt_id = owner.default_analysis
+            sql = """INSERT INTO qiita.{0}
+                    (email, name, description, analysis_status_id)
+                    VALUES (%s, %s, %s, 3)
```
Comment (josenavas) [inline qiita_db/analysis.py]: can we avoid hardcoding IDs? We never now when these can change. Using the `convert_to_id` function should make the trick.

File: qiita_db/analysis.py [hunk 2]
```diff
@@ -112,23 +112,44 @@ def create(cls, owner, name, description, parent=None):
...
+                                                   description))
+            sql = """UPDATE qiita.analysis_sample
+                     SET analysis_id = %s
+                     WHERE analysis_id = %s RETURNING %s"""
+            conn_handler.add_to_queue(queue, sql, ['{0}', dflt_id, '{0}'])
+        else:
+            # insert analysis information into table as "in construction"
+            sql = """INSERT INTO qiita.{0}
+                  (email, name, description, analysis_status_id)
+                  VALUES (%s, %s, %s, 1)
```
Comment (josenavas) [inline qiita_db/analysis.py]: Same here.

**Comment by josenavas** [comment]
Just a small comment.

File: qiita_db/analysis.py (hunk 1 above)
Comment (squirrelo) [inline qiita_db/analysis.py]: convert_to_id doesn't currently work with status tables, hence issue #292. I have updated the `convert_to_id` function to work around the issue for now though.

Comment (squirrelo) [inline qiita_db/analysis.py]: Aaaand that update breaks all the things. Using a specific SQL statement here and tagged with issue ID.

Comment (josenavas) [inline qiita_db/analysis.py]: Thank you.

**Comment by squirrelo** [comment]
Changes made and ready for another review.

File: qiita_db/analysis.py [hunk 3]
```diff
@@ -112,23 +112,53 @@ def create(cls, owner, name, description, parent=None):
...
+            dflt_id = owner.default_analysis
+            sql = """INSERT INTO qiita.{0}
+                    (email, name, description, analysis_status_id)
+                    VALUES (%s, %s, %s, %s)
+                    RETURNING analysis_id""".format(cls._table)
+            conn_handler.add_to_queue(queue, sql, (owner.id, name,
+                                                   description, status_id))
+            # MAGIC NUMBER 3: command selection step
+            # needed so we skip the sample selection step
+            sql = """INSERT INTO qiita.analysis_workflow
```
Comment (josenavas) [inline qiita_db/analysis.py]: I just noticed this. Why is the `analysis_workflow` table needed? Is it only handling a column called step, and there is no further control on it, so it looks like that column can be included in the analysis object???

**Comment by josenavas** [comment]
@squirrelo just one comment that should not block the PR, but I'd like to get an answer to check if we actually need to open an issue for that, as it looks like that table is not needed at all.
Thanks!

File: qiita_db/analysis.py (hunk 3 above)
Comment (squirrelo) [inline qiita_db/analysis.py]: The table was supposed to be a housekeeping table to track what step the analysis is on and any other things that needed tracking during analysis creation. I guess it turns out we didn't need as much housekeeping as originally thought.

Comment (josenavas) [inline qiita_db/analysis.py]: Yeah, I think we are keeping track of everything else in the multiple tables that build up the analysis and the step just limits what can be done on the analysis. I've created a new issue https://github.com/biocore/qiita/issues/1052 and I assigned to beta. Later we can decide to exactly which alpha release it belongs, but I don't think it is either 0.1 or 0.2. Thanks for the feedback!

**Comment by josenavas** [comment]
:+1:

**Comment by squirrelo** [comment]
Can this get a second reviewer?

**Comment by ElDeveloper** [comment]
I'll do this in a bit.

**Pull request merged**